docs/sources/experiments/cohort-labels.md

```diff
@@ -0,0 +1,493 @@
+
+The entity id is the same as the cohort above: it identifies the restaurant. The date is just the date that the inspection happened, and the result is a string 'pass'/'fail' stating whether or not the restaurant passed the inspection. 
+
+
+#### Inspections Label Config
+
+In constructing the label query, we have to consider the note above that we want to return only one row for a given entity id. The easiest way to do this, given that this query is run per as-of-date, is to group by the entity id and aggregate all the matched events somehow. In this case, a sensible definition is that we want any failed inspections to trigger a positive label. So if there is one pass and one fail that falls under the label timespan , the label should be True. `bool_or` is a handy Postgres aggregation function that does this.
+
+A query to find any failed inspections would be written in an experiment YAML config as follows:
+
+```yaml
+label_config:
+  query: |
+    select
+    entity_id,
+    bool_or(result = 'fail')::integer as outcome
+    from inspections
+    where '{as_of_date}'::timestamp <= date
+    and date < '{as_of_date}'::timestamp + interval '{label_timespan}'
+    group by entity_id
+  name: 'failed_inspection'
+```
+
+### Example: Early Intervention
+
+#### Early Intervention Label Source Table
+
+We reuse the [generic events table](#early-intervention-cohort-source-table) used in the early intervention cohort section.
+
+#### Early Intervention Label Config
+
+We would like to assign a `True` label to everybody who is booked into jail within the label timespan. Note the `include_missing_labels_in_train_as` value: `False`. Anybody who does not show up in this query can be assumed to not have been booked into jail, so they can be assigned a `False` label.
+
+
+```
+label_config:
+    query: |
+        SELECT entity_id,
+               bool_or(CASE WHEN event_type = 'booking' THEN TRUE END)::integer AS outcome
+          FROM events
+         WHERE knowledge_date <@ daterange('{as_of_date}'::date, ('{as_of_date}'::date + interval '{label_timespan}')::date)
+         GROUP BY entity_id
+    include_missing_labels_in_train_as: False
+    name: 'booking'
+```
+
+### Example: Visits
+
+#### Visits Label Source Table
+
+We reuse the [diabetes_diagnoses table](#visits-cohort-source-tables) from the cohort section.
+
+#### Visits Label Config
+
+We would like to identify people who are diagnosed with diabetes within a certain `label_timespan` after the given `as-of-date`. Note that `include_missing_labels_in_train_as` is False here as well. Any diagnoses would show up here, so the lack of any results from this query would remove all ambiguity.
+
+```
+label_config:
+  query: |
+    select entity_id, 1 as outcome
+    from diabetes_diagnoses
+    where as_of_date <@ daterange('{as_of_date}' :: date, ('{as_of_date}' :: date + interval '{label_timespan}') :: date)
+    group by entity_id
+  include_missing_labels_in_train_as: False
+  name: 'diabetes'
+```
```

Comment:
Might be a good example for a comment about being thoughtful about your label definitions since an alternative way to think about this problem would be something more similar to the inspections problem (e.g., you would need to run an A1C test to either diagnosis someone with diabetes or positively conclude they don't have it, so anyone who hasn't been tested would be a null in that case...)